### DISCUSSION TITLE
Have you reconsidered adding WARC support?

### DISCUSSION THREAD

**Discussion by YousufSSyed** [gildas-lormeau/SingleFile#1418] [Q&A]
I saw [the issue for it](https://github.com/gildas-lormeau/SingleFile/issues/160) posted all the way back in 2019 and I think its a really good time to look at supporting the WARC format.
* There's a lot more software that supports it now.
* It can be viewed in the browser with sites like https://replayweb.page.
* WARCs (both `.warc` and `.warc.gz`) can easily be concatenated, with unix `cat` in the command line for instance.
* When combined, their resources can be deduplicated, allowing space to be saved.
* Can be added to Web replay software like [Pywb](https://github.com/webrecorder/pywb).
* There's currently no good addons to download pages as WARCs, Warcreate is only available on Chrome and not Firefox. Other software requires lots of setup outside of the browser.

**Comment by gildas-lormeau** [comment]
The fundamental problem is that SingleFile does not inspect network exchanges, whereas the WARC format was designed to do just that. That's why I haven't delved into the subject.

**Comment by ibnesayeed** [comment]
I still think there is a pathway to support WARC format and a potential benefit associated. The entire bundled single file can be treated as the representation of the primary URL it was initiated from and can be put in a single WARC record. The WARC record would allow adding more metadata in headers and the WARC file can then be ingested into archival playback systems to serve these captures along with other traditional WARCs.

**Comment by github-account1111** [comment]
Why use WARC over HTML? Sounds like you sacrifice a good amount of compatibility (HTML can even be opened on anything with a browser) for very modest space savings (since it seems WARC contents are individually compressed)?

It also seems you lose the faithfulness aspect of SingleFile with WARC. I can't recall ever viewing a WayBack Machine-archived copy of a page that wasn't garbled in some way.

**Reply by pirate** [reply]
The whole point would be that singlefile's output is great and *not garbled* because it's using a full browser, and WARC-ifying it allows easy interop with other archiving tools. For example it would allow users to save nice singlefile snapshots from their browser, then upload them to archive.org and view them there.
